src/components/ComponentSwitch/ComponentSwitch.tsx

```diff
@@ -1,37 +1,106 @@
+import { Box, ListItem, ListItemProps } from '@chakra-ui/react'
 import { PresentationslidesUnion } from '@lib/generated/sdk'
+import { Dispatch, ReactNode, SetStateAction } from 'react'
 //Components
-import SlideBasic from './Slides/Slidebasic'
-import SlideImage from './Slides/SlideImage'
-import SlideImageFull from './Slides/SlideImageFull'
-import SlidePlanning from './Slides/SlidePlanning'
+import { SlideBasic, SlideBasicThumbnail } from './Slides/Slidebasic'
+import { SlideImage, SlideImageThumbnail } from './Slides/SlideImage'
+import {
+	SlideImageFull,
+	SlideImageFullThumbnail,
+} from './Slides/SlideImageFull'
+import { SlidePlanning, SlidePlanningThumbnail } from './Slides/SlidePlanning'
+import slideStore from './../../state/slide'
 
 interface ComponentSwitchProps {
-	data: PresentationslidesUnion[]
+	data: PresentationslidesUnion[] | PresentationslidesUnion
 }
 
 export function ComponentSwitch(props: ComponentSwitchProps) {
 	const { data } = props
+	const { activeIndex } = slideStore()
 
 	if (!data) return null
 
 	return (
 		//TODO: Built slide components
 		<>
-			{data.map(slide => {
-				switch (slide.__typename) {
-					case 'SlideBasic':
-						return <SlideBasic key={`slidebasic-${slide.id}`} />
-					case 'SlideImage':
-						return <SlideImage key={`slideimage-${slide.id}`} />
-					case 'SlideImageFull':
-						return <SlideImageFull key={`slideimagefull-${slide.id}`} />
-					case 'SlidePlanning':
-						return <SlidePlanning key={`slideplanning-${slide.id}`} />
-					default:
-						console.log('Component not found', slide)
-						return null
-				}
-			})}
+			{Array.isArray(data) ? (
```

Comment:
Array is a too generic name

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray